websites/S/Spyke/presence.ts

```diff
@@ -0,0 +1,96 @@
+				JSON.parse(localStorage.getItem("persist:Token_Data_Persist"))
+					.jwtIdToken
+			)}`,
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify({
+			query,
+			variables: vars,
+			operationName,
+		}),
+	});
+	return response.json();
+}
+
+let url = "https://spyke.social/";
+
+presence.on("UpdateData", () => {
+	const nurl = document.URL;
+	if (nurl !== url) {
+		const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/v10pkZA.png",
+			smallImageKey: "https://i.imgur.com/55dCzB1.png",
+			details: "A Community Discussion Platform",
+			state: "On Homepage",
+			startTimestamp: browsingTimestamp,
+		};
+		presenceData.buttons = [
+			{ label: "View Spyke", url: "https://spyke.social/" },
+		];
+
+		if (document.URL.includes("/p/")) {
+			postGQLAPI(
+				"presencepst",
+				"query presencepst($id: ID!) { comments(ids: [$id]) { ... on Post { title content { id data __typename } communities { name dp } } } }",
+				{ id: document.URL.split("/p/")[1] }
+			).then(res => {
+				if (res.data.comments[0].content[0].__typename === "Image") {
+					presenceData.largeImageKey = res.data.comments[0].content[0].data;
+					presenceData.smallImageKey = res.data.comments[0].communities[0].dp;
+					presenceData.details = `Reading post in ${res.data.comments[0].communities[0].name}`;
+					presenceData.state = res.data.comments[0].title;
+					presenceData.buttons = [{ url: document.URL, label: "View Post" }];
+					presence.setActivity(presenceData);
+				} else {
+					presenceData.largeImageKey = res.data.comments[0].communities[0].dp;
+					presenceData.smallImageKey = "https://i.imgur.com/v10pkZA.png";
+					presenceData.details = `Reading post in ${res.data.comments[0].communities[0].name}`;
+					presenceData.state = res.data.comments[0].title;
+					presenceData.buttons = [{ url: document.URL, label: "View Post" }];
+					presence.setActivity(presenceData);
+				}
+			});
+		} else if (document.URL.includes("/g/")) {
+			postGQLAPI(
+				"presencegrp",
+				"query presencegrp($name:String!){ communityByName(name:$name) { name dp cover}}",
+				{ name: document.URL.split("/g/")[1] }
+			).then(res => {
+				presenceData.largeImageKey = res.data.communityByName.cover;
+				presenceData.smallImageKey = res.data.communityByName.dp;
+				presenceData.state = `${res.data.communityByName.name}`;
+				presenceData.details = "Viewing a Clan";
+				presenceData.buttons = [{ url: document.URL, label: "View Clan" }];
+				presence.setActivity(presenceData);
+			});
+		} else if (document.URL.includes("/upload")) {
+			presenceData.largeImageKey = "https://i.imgur.com/v10pkZA.png";
+			presenceData.smallImageKey = "https://i.imgur.com/55dCzB1.png";
+			presenceData.details = "Uploading a post";
+			presenceData.state = "";
+			presence.setActivity(presenceData);
+		} else if (document.URL.includes("/u/")) {
+			presenceData.largeImageKey = "https://api.dicebear.com/7.x/avataaars/png"; //`https://i.imgur.com/v10pkZA.png`;
```

Comment:
Assets should be 512x512px. Also, since assets are used many times in this presence, place them in an enum like so:
https://github.com/PreMiD/Presences/blob/1f5a4117f92c6549a75e1c01c6305f9c4398b5b5/websites/A/Aodesu/presence.ts#L6-L14